### PR TITLE
Support for Custom Resolver with Location as optional

### PR DESCRIPTION
**Pull request by DwitikrDash**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Comment by DwitikrDash** [comment]
@kavya498 Can you please review this

**Comment by kavya498** [comment]
@DwitikrDash , please attach testcase results

File: ibm/resource_ibm_private_dns_custom_resolver.go [hunk 1]
```diff
@@ -169,13 +168,7 @@ func resouceIBMPrivateDNSCustomResolverCreate(context context.Context, d *schema
 	d.SetId(convertCisToTfTwoVar(*result.ID, crn))
 	d.Set(pdnsCRId, *result.ID)
 
-	_, err = waitForPDNSCustomResolverHealthy(d, meta)
```
Comment (kavya498) [inline ibm/resource_ibm_private_dns_custom_resolver.go:172]: why is this wait logic removed?

File: ibm/resource_ibm_private_dns_custom_resolver.go [hunk 2]
```diff
@@ -155,11 +154,11 @@ func resouceIBMPrivateDNSCustomResolverCreate(context context.Context, d *schema
 		crDescription = des.(string)
 	}
 
-	crLocations := d.Get(pdnsCustomResolverLocations).(*schema.Set)
+	// crLocations := d.Get(pdnsCustomResolverLocations).(*schema.Set)
```
Comment (kavya498) [inline ibm/resource_ibm_private_dns_custom_resolver.go:157]: location field is made optional..
that means user can stll give location.. please use d.getok() and then set values in customResolverOption

File: ibm/resource_ibm_private_dns_custom_resolver_location_test.go [hunk 3]
```diff
@@ -14,8 +14,8 @@ import (
 func TestAccIBMPrivateDNSCRLocations_Basic(t *testing.T) {
 	name := fmt.Sprintf("testpdnscustomresolver%s", acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum))
 	description := "new test CR Locations - TF"
-	subnet_crn := "crn:v1:staging:public:is:us-south-1:a/01652b251c3ae2787110a995d8db0135::subnet:0716-03d54d71-b438-4d20-b943-76d3d2a1a590"
-	subnet_crn_new := "crn:v1:staging:public:is:us-south-1:a/01652b251c3ae2787110a995d8db0135::subnet:0716-84dcb64e-3ada-45fd-b0f7-94de1ac0d16b"
+	subnet_crn := "crn:v1:staging:public:is:us-south-2:a/01652b251c3ae2787110a995d8db0135::subnet:0726-b6f3cb83-48f0-4c55-9023-202fe4570c83"
```
Comment (kavya498) [inline ibm/resource_ibm_private_dns_custom_resolver_location_test.go:17]: please update testcases.. donot hardcode values

File: website/docs/r/dns_custom_resolver.html.markdown [hunk 4]
```diff
@@ -43,11 +39,6 @@ Review the argument reference that you can specify for your resource.
 - `instance_id` - (Required, String) The GUID of the private DNS service instance.
 - `name`- (Required, String) The name of the custom resolver.
 - `description` - (Optional, String) Descriptive text of the custom resolver.
-- `locations`- (Required, Set) The list of locations where this custom resolver is deployed. There is no update for location argument in resolver resource.
```
Comment (kavya498) [inline website/docs/r/dns_custom_resolver.html.markdown:46]: please donot remove.. it is made as optional.. feature is not removed from resource

File: ibm/resource_ibm_private_dns_custom_resolver_location.go [hunk 5]
```diff
@@ -95,6 +97,17 @@ func resourceIBMPrivateDNSLocationCreate(context context.Context, d *schema.Reso
 		return diag.FromErr(fmt.Errorf("Error creating the custom resolver location %s:%s", err, resp))
 	}
 	d.SetId(convertCisToTfThreeVar(*result.ID, resolverID, instanceID))
+	_, err = waitForPDNSCustomResolverHealthy(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	optCr := sess.NewUpdateCustomResolverOptions(instanceID, resolverID)
+	optCr.SetEnabled(true)
+	resultCr, respCr, errCr := sess.UpdateCustomResolverWithContext(context, optCr)
```
Comment (kavya498) [inline ibm/resource_ibm_private_dns_custom_resolver_location.go:106]: it should be enabled only if user sets enabled to true.. we cannot hardcode it..

File: ibm/resource_ibm_private_dns_custom_resolver.go (hunk 1 above)
Comment (DwitikrDash) [reply ibm/resource_ibm_private_dns_custom_resolver.go:172]: Wait logic added - It is triggered if location is provided

File: website/docs/r/dns_custom_resolver.html.markdown (hunk 4 above)
Comment (DwitikrDash) [reply website/docs/r/dns_custom_resolver.html.markdown:46]: Updated documents

File: ibm/resource_ibm_private_dns_custom_resolver_location.go (hunk 5 above)
Comment (DwitikrDash) [reply ibm/resource_ibm_private_dns_custom_resolver_location.go]: Added check to see if enable is set before enabling

**Comment by hkantare** [comment]
I think this was made already optional with another PR

**Pull request closed without merging**